thirdparty/azure-sdk-cpp/fix-openssl-helper.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/sdk/attestation/azure-security-attestation/src/private/crypto/openssl/openssl_helpers.hpp b/sdk/attestation/azure-security-attestation/src/private/crypto/openssl/openssl_helpers.hpp
+index a52791cea..b5e76a0f9 100644
+--- a/sdk/attestation/azure-security-attestation/src/private/crypto/openssl/openssl_helpers.hpp
++++ b/sdk/attestation/azure-security-attestation/src/private/crypto/openssl/openssl_helpers.hpp
+@@ -39,7 +39,7 @@ namespace Azure { namespace Security { namespace Attestation { namespace _detail
+   using openssl_evp_pkey = openssl_unique_ptr<EVP_PKEY>;
+   using openssl_bio = openssl_unique_ptr<BIO>;
+ 
+-#ifdef __cpp_nontype_template_parameter_auto
++#if defined(__cpp_nontype_template_parameter_auto) && !defined(_WIN32)
```

Comment:
can you add a comment about why we need to exclude msvc here, please?

Reply:
Added comment in https://github.com/apache/nifi-minifi-cpp/pull/2005/commits/83360da0319638c7cf4b1198742f5f5ad163a1f5

Reply:
After discussing and checking this issue further in private, this seems to be a bug in the C++17 implementation/usage of the `make_openssl_unique` helper function in the Azure SDK. On Windows we override the `/std:c++14` flag set in the build files with `/std:c++latest` so we use the faulty code there. I opened an issue for this on the Azure SDK project page https://github.com/Azure/azure-sdk-for-cpp/issues/6697 and patched this issue in the latest commit: https://github.com/apache/nifi-minifi-cpp/pull/2005/commits/380d04246a5f496bcfb67ce3ccad65af6f1cbec5